### PR TITLE
Add proposal for delivery interface messages

### PR DESCRIPTION
**Pull request by Sushant-Chavan**
Pull request to partially handle issue #32

Adds docs/messages/mailbox.md containing the input and output messages

### CODE REVIEW

File: docs/mailbox.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Message to mailbox
+### The TruckAgent informs the mailbox about the delivered order
+#### String: "OrderID"
+<br>
+
+
+## Message from the mailbox
+### The mail box informs the concerned customer and other agents who are interested in knowing the delivery status of the order
+#### String: "OrderID"
```
Comment (argenos) [inline docs/mailbox.md]: What about the time it was delivered?

File: docs/mailbox.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## Message to mailbox
+### The TruckAgent informs the mailbox about the delivered order
+#### String: "OrderID"
+<br>
```
Comment (argenos) [inline docs/mailbox.md]: A couple of things that are missing that I can think of is the number of boxes and which bakery sent it. Remember a single client may be receiving orders from different bakeries.

Comment (Sushant-Chavan) [reply docs/mailbox.md]: This has been fixed with the new commit

File: docs/mailbox.md (hunk 1 above)
Comment (Sushant-Chavan) [reply docs/mailbox.md]: The time and date of delivery is also provided in the updated message description

**Comment by Sushant-Chavan** [comment]
@argenos we have fixed your review findings in the new commit. If you need some more changes to it, please let us know.

**Comment by argenos** [comment]
I requested some minor changes, mostly formatting.

File: docs/mailbox.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+## Message to mailbox
```
[the remark this answers is not in the document]
Comment (Sushant-Chavan) [reply docs/mailbox.md]: Done

File: docs/mailbox.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+### The TruckAgent informs the mailbox about the delivered order
```
[the remark this answers is not in the document]
Comment (Sushant-Chavan) [reply docs/mailbox.md]: Done

File: docs/mailbox.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+#### ACL Performative: INFORM
```
[the remark this answers is not in the document]
Comment (Sushant-Chavan) [reply docs/mailbox.md]: Done

File: docs/mailbox.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+#### Message Content: JSON string with the content described below
```
[the remark this answers is not in the document]
Comment (Sushant-Chavan) [reply docs/mailbox.md]: Done

File: docs/mailbox.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    "DeliveredBy": "String-TruckID",
+    "DayOfDelivery": "Integer-Day(0-365)",
+    "TimeOfDelivery": "Integer-HourOfDay(0-23)",
+    "NumOfBoxes": "Integer-CountOfBoxes",
+    "ProducedBy": "String-NameOfBakery"
+  }
+}
+```
+
+#### Example message:
```
[the remark this answers is not in the document]
Comment (Sushant-Chavan) [reply docs/mailbox.md]: Done

File: docs/mailbox.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+```json
+{
+  "DeliveryStatus": {
+    "OrderID": "Order-152",
+    "DeliveredTo": "Customer-001",
```
[the remark this answers is not in the document]
Comment (Sushant-Chavan) [reply docs/mailbox.md]: Removed the _OrderID_ field.

Updated the name of the field _DeliveredTo_ to _OrderDeliveredTo_

File: docs/mailbox.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    "NumOfBoxes": 5,
+    "ProducedBy": "Flying Saucers Bakery"
+  }
+}
+```
+
+<br>
+
+
+## Message from the mailbox
```
[the remark this answers is not in the document]
Comment (Sushant-Chavan) [reply docs/mailbox.md]: Done

**Comment by Sushant-Chavan** [comment]
@argenos We made the formatting changes suggested by you. Please have a look at the changes

**Pull request merged**